Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -53,6 +53,22 @@ public typealias StartPurchaseBlock = (@escaping PurchaseCompletedBlock) -> Void
  *  framework handle the singleton instance for you.
  */
 @objc(RCPurchases) public final class Purchases: NSObject, PurchasesType, PurchasesSwiftType {
+    public func purchase(product: StoreProduct, completion: @escaping PurchaseCompletedBlock) {
+        purchasesOrchestrator.purchase(product: product, package: nil, transactionMetadata: nil, completion: completion)
+    }
+
+    public func purchase(package: Package, completion: @escaping PurchaseCompletedBlock) {
+        purchasesOrchestrator.purchase(product: package.storeProduct, package: nil, transactionMetadata: nil, completion: completion)
+    }
+
+    public func purchase(product: StoreProduct, transactionMetadata: [String : String]?, completion: @escaping PurchaseCompletedBlock) {
+        purchasesOrchestrator.purchase(product: product, package: nil, transactionMetadata: transactionMetadata, completion: completion)
+    }
+
+    public func purchase(package: Package, transactionMetadata: [String : String]?, completion: @escaping PurchaseCompletedBlock) {
```

Comment:
Instead of making brand new purchase functions to make a purchase with metadata, I'd consider modifying the existing purchase functions and making the `transactionMetadata` parameter default to `nil` so that it isn't a breaking change.

There are quite a few purchase functions today across both `Purchases.swift` and `Purchases+async.swift`. Ideally folks would be able to upload transaction metadata no matter which function call they're already using without having to migrate to a new one.